### PR TITLE
Upgrade to Fastify v3

### PR DESCRIPTION
**Pull request by thomheymann**
Upgrade to Fastify v3

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
Node 6.x and Node 8.x must be dropped.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        node-version: [6.x, 8.x, 10.x, 11.x, 12.x, 13.x]
+        node-version: [10.x, 11.x, 12.x, 13.x, 14.x]
```
Comment (mcollina) [inline .github/workflows/ci.yml:8]:
```suggestion
        node-version: [10.x, 12.x, 14.x]
```

File: index.d.ts [hunk 2]
```diff
@@ -1,14 +1,22 @@
-import { Plugin } from 'fastify';
-import { IncomingMessage, Server, ServerResponse } from 'http';
+import { FastifyPlugin, FastifyReply, RawServerBase } from 'fastify';
 
 declare module "fastify" {
-  interface FastifyReply<HttpResponse> {
-    view(page: string, data?: object): FastifyReply<HttpResponse>;
+  interface FastifyReplyInterface {
+    view(page: string, data?: object): FastifyReply<RawServerBase>;
```
Comment (Ethan-Arrowood) [inline index.d.ts:5]: generic shouldn't be necessary

File: package.json [hunk 3]
```diff
@@ -43,7 +43,7 @@
     "ejs": "^3.0.1",
     "ejs-mate": "^2.0.0",
     "express": "^4.17.1",
-    "fastify": "^2.12.0",
+    "fastify": "^3.0.0-alpha.1",
```
Comment (Ethan-Arrowood) [inline package.json:46]: use `rc`

**Pull request closed without merging**